internal/docker/deployment.go

```diff
@@ -57,6 +57,16 @@ func (hsDep *HomeserverDeployment) SetEndpoints(baseURL string, fedBaseURL strin
 	}
```

Comment:
> Do you know what those CI failures are about? The error doesn't seem related to these changes.
>
> *@devonh, https://github.com/matrix-org/complement/pull/778#issuecomment-2881802935*

No idea :shrug:

Reply:
Hmm seems like we may benefit from changing any use of `client.NewEnvClient` to 
```go
cli, err := client.NewClientWithOpts(
	client.FromEnv,
	client.WithAPIVersionNegotiation(),
)
```
https://pkg.go.dev/github.com/docker/docker/client#NewClientWithOpts

I think that may avoid this intermittent sounding issue with docker versioning altogether.

Reply:
I think there are 2 locations of this, in builder.go & deployer.go

Reply:
:-1: Was already on it (changed both places)

Reply:
CI is green :white_check_mark: 

Great shout on the fix @devonh!